src/Mvc/Mvc.Core/src/ApplicationModels/InferParameterBindingInfoConvention.cs

```diff
@@ -132,4 +139,28 @@ private bool IsComplexTypeParameter(ParameterModel parameter)
 
         return metadata.IsComplexType;
     }
+
+    private bool IsOptionalParameter(ParameterModel parameter)
+    {
+        if (parameter.ParameterInfo.HasDefaultValue)
+        {
+            return true;
+        }
+
+        if (_modelMetadataProvider is ModelMetadataProvider modelMetadataProvider)
+        {
+            var metadata = modelMetadataProvider.GetMetadataForParameter(parameter.ParameterInfo);
+            return !metadata.IsRequired;
+        }
+        else
+        {
+            // Cannot be determine if the parameter is optional since the provider
+            // does not provides an option to getMetadata from the parameter info
+            // so, we will use the Nullability context
+
+            // No need for information from attributes on the parameter. Just use its type.
+            var metadata = _modelMetadataProvider.GetMetadataForType(parameter.ParameterInfo.ParameterType);
+            return metadata.NullabilityState != NullabilityState.NotNull;
```

Comment:
Do we want to infer optionality for `NullabilityState.Unkown`?

Reply:
That is basically how the IsRequired is set and how I documented in the issue.

The idea is If the user wants a required body he needs to set the Required attribute otherwise it is optional when the context is unknown. Make sense?

Reply:
I see this does align with the minimal route handler behavior. I thought we were more conservative about passing in null when nullable annotations were disabled, but I guess not.

On one hand, I do like making minimal and mvc consistent, but on the other hand this does seem like a dangerous breaking change to make without making it opt-in. Having the following action going from rejecting empty bodies to allowing null could affect the business logic in unforeseen ways.


```C#
public IActionResult Action([FromBody] string value);
```

In many cases, it might lead to a quick NullReferenceException and a 500 response which wouldn't be so bad, but I could also see it allowing invalid data to be persisted. This behavior only makes a difference when nullable annotations are disabled, so I think I can live with the inconsistency as nullable annotations are becoming more and more common.

Reply:
After offline discussion, I think it's very unlikely someone would rely on `[FromBody]` rather `[Required]` to get this behavior. It's better to be consistent with the non-attributed inferred from body behavior. I'm happy with leaving it as is without an option.